_ml-commons-plugin/gpu-acceleration.md

```diff
@@ -0,0 +1,285 @@
+
+## Supported images
+
+GPU acceleration supports both Docker images and [Amazon Machine Images (AMI)](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html).
+
+## Setting up a GPU accelerated ML node
+
+Depending on the GPU, you can provision an GPU accelerated ML node using automated initialization scripts or manually. 
+
+### Mount an nvidia-uvm device
+
+NVIDIA uses CUDA®, NVIDIA's programming platform, to increase node performance. In order to take advantage of CUDA, you need to make sure that your drivers contain the `nvidia-uvm` kernels inside the `/dev` directory. To check for the kernels, enter `ls -al /dev | grep nvidia-uvm`.
+
+If the `nvidia-uvm` does not exist, run the `nvidia-uvm-init.sh` script. 
+
+```
+#!/bin/bash
+## Script to initialize nvidia device nodes.
+## https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#runfile-verifications
+/sbin/modprobe nvidia
+if [ "$?" -eq 0 ]; then
+  # Count the number of NVIDIA controllers found.
+  NVDEVS=`lspci | grep -i NVIDIA`
+  N3D=`echo "$NVDEVS" | grep "3D controller" | wc -l`
+  NVGA=`echo "$NVDEVS" | grep "VGA compatible controller" | wc -l`
+  N=`expr $N3D + $NVGA - 1`
+  for i in `seq 0 $N`; do
+    mknod -m 666 /dev/nvidia$i c 195 $i
+  done
+  mknod -m 666 /dev/nvidiactl c 195 255
+else
+  exit 1
+fi
+/sbin/modprobe nvidia-uvm
+if [ "$?" -eq 0 ]; then
+  # Find out the major device number used by the nvidia-uvm driver
+  D=`grep nvidia-uvm /proc/devices | awk '{print $1}'`
+  mknod -m 666 /dev/nvidia-uvm c $D 0
+  mknod -m 666 /dev/nvidia-uvm-tools c $D 0
+else
+  exit 1
+fi
+```
+
+After verification that `nvidia-uvm` exists under `/dev`, you can start OpenSearch inside your cluster. 
+
+### Prepare AWS Inferentia
+
+Depending on the Linux operating system running on AWS Inferentia, use the follow commands and scripts to set provision an ML node and get OpenSearch running inside your cluster.
+
+Download OpenSearch and install in your Linux instance's home folder.
+
+```
+cd ~; wget https://artifacts.opensearch.org/releases/bundle/opensearch/2.5.0/opensearch-2.5.0-linux-x64.tar.gz
+tar -xvf opensearch-2.5.0-linux-x64.tar.gz
+
+echo "export OS_HOME=~/opensearch-2.5.0" | tee -a ~/.bash_profile
+echo "export PYTORCH_VERSION=1.12.1" | tee -a ~/.bash_profile
+source ~/.bash_profile
+```
+
+Next, create a shell script file called `prepare_torch_neuron.sh`. Use the following examples based on your Linux operating system. After you've run the scripts, exit your current terminal and open a new terminal to start OpenSearch.
+
+GPU acceleration has only been tested on Ubuntu 20.04 and Amazon Linux2. However, other you can use other Linux operating systems.
+{: .note}
+
+#### Ubuntu 20.04
+
+```
+. /etc/os-release
```

Comment:
Break this out into individual steps instead of a code block since they are sequential commands that need to be run manually.

Reply:
Also...What is `. /etc/os-release`?  Is that supposed to have a `cat` in front?

Reply:
In AWS Inferentia, it contains environment variables for the Inferentia instance. This line pulls those variables for use in different parts of the script.